client/src/pages/SignUp.js

```diff
@@ -0,0 +1,278 @@
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import Background from '../elements/Background';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(20),
+    padding: theme.spacing(5),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  },
+  headerText: {
+    marginTop: theme.spacing(10)
+  },
+  form: {
+    width: '100%',
+    marginTop: theme.spacing(3),
+    textAlign: 'center'
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+    padding: theme.spacing(1, 6),
+    border: 0,
+    borderRadius: 25,
+    color: '#fff',
+    textTransform: 'none',
+    fontSize: 15
+  }
+}));
+
+export default function SignIn() {
+  const classes = useStyles();
+
+  // ========== FORM VALIDATION LOGIC BEGIN ==========
+
+  // Validate username
+  const [ username, setUsername ] = useState({
+    value: '',
+    error: 'Field can not be blank',
+    touched: false
+  });
+
+  const usernameHandler = (event) => {
+    const usernameRegex = /^[a-zA-Z0-9-]+$/;
+    let errorFound = '';
+    if (event.target.value.length < 4 || event.target.value.length > 20) {
+      errorFound = 'Username length should be 4-20 chars long';
+    } else if (!event.target.value.match(usernameRegex)) {
+      errorFound = 'Only letters, numbers and  "-" are  acceptable';
+    }
+    setUsername({
+      value: event.target.value,
+      touched: true,
+      error: errorFound
+    });
+  };
+
+  // Validate email
+  const [ email, setEmail ] = useState({
+    value: '',
+    error: 'Field can not be blank',
+    touched: false
+  });
+
+  const emailHandler = (event) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    let errorFound = '';
+    if (!event.target.value.match(emailRegex)) {
+      errorFound = 'Incorrect email address';
+    }
+    setEmail({
+      value: event.target.value,
+      touched: true,
+      error: errorFound
+    });
+  };
+
+  //Validate password
+  const [ password, setPassword ] = useState({
+    value: '',
+    error: 'Field can not be blank',
+    touched: false
+  });
+
+  const passwordHandler = (event) => {
+    let errorFound = null;
+
+    switch (true) {
+      case event.target.value.length < 6:
+        errorFound = 'Password must be at least 6 characters long';
+        break;
+      case !event.target.value.match(/(?=.*[a-z])/):
+        errorFound = 'Password must contain at least one lower case letter';
+        break;
+      case !event.target.value.match(/(?=.*\d)/):
+        errorFound = 'Password must contain at least one digit';
+        break;
+      case !event.target.value.match(/(?=.*[A-Z])/):
+        errorFound = 'Password must contain at least one upper case letter';
+        break;
+      case !event.target.value.match(/(?=.*[!@#$%^&*])/):
+        errorFound = 'Password must contain at least one special char';
+        break;
```

Comment:
It looks like this can be simplified into a single regex to solve this password validation.

Reply:
I think this is a bit too involved of a password validator - I like that you did this, but I'm a fan of just doing length checks, especially for these projects so it's easy for people to create accounts. There is a whole debate on if password validators should be hard or easy, but I think for this project, we want to make them easy

Reply:
@svitaly I added an icebox ticket #23 with the assumption that hard password requirements would require guest account creation to give users that are reviewing the app easy access. Maybe stash that validator and follow @skassam21 suggestion to make it length only for now.

Reply:
ok, will move vaildator to utils.